File: oblig2/Lege.java
/*
  Klasse for leger
*/
public class Lege {

    // Variabler
    protected String navn;

    // Konstruktør
    public Lege(String legensNavn) {
	navn = legensNavn;
    }

    // Getters
    public String hentNavn() {
	return navn;
    }

    public String toString() {
	return "Lege " + hentNavn();
    }
}

File: oblig2/cpp/Lege.cpp
#include "Lege.h"

Lege::Lege(const char* navn) : m_navn(std::move(navn)) { };
const char* Lege::hentNavn() const
{
  return m_navn;
}

Spesialist::Spesialist(const char* name, int kontrollID)
  : Lege(std::move(name)), m_kontrollID(kontrollID)
{ }

int Spesialist::hentKontrollID()
{
  return m_kontrollID;
}


std::ostream& operator<<(std::ostream& out, const Lege& L)
{
  return out << "Lege " << L.hentNavn();
}
std::ostream& operator<<(std::ostream& out, const Spesialist& S)
{
  return out << "Spesialist " << (Lege) S;
}
